src/Features/CSharp/Portable/InlineDeclaration/CSharpInlineDeclarationCodeFixProvider.cs

```diff
@@ -49,17 +50,26 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
             // of the local.  This is necessary in some cases (for example, when the
             // type of the out-var-decl affects overload resolution or generic instantiation).
 
-            foreach (var diagnostic in diagnostics)
+            var originalRoot = editor.OriginalRoot;
+
+            var originalNodes = new (ILocalSymbol, IdentifierNameSyntax, DeclarationExpressionSyntax, SyntaxNode)[diagnostics.Length];
+
+            for (var i = 0; i < originalNodes.Length; i++)
             {
-                cancellationToken.ThrowIfCancellationRequested();
-                await AddEditsAsync(
-                    document, editor, diagnostic, 
-                    options, cancellationToken).ConfigureAwait(false);
+                originalNodes[i] = await GetDeclarationAsync(document, editor, diagnostics[i], options, cancellationToken).ConfigureAwait(false);
```

Comment:
I was going to suggest using `Task.WhenAll` to avoid the long declaration of `originalNodes` then I realized `GetDeclarationAsync` has side-effects. Maybe just call it something like `AddInlineWithVarEditsAsync` to make it more clear what it actually does? Sorry, I'm terrible at naming things, but I'm sure you know I'm saying :)

Reply:
SyntaxEditor likely should *not* be passed into GetDeclarationAsync.